### ISSUE TITLE
Webpack configurations as NPM modules

### ISSUE THREAD

**Issue by Merott** [preboot/angular2-webpack#202]
The one common problem I find with most Angular starters is the difficulty in updating the configurations and the core functionality that they provide, once you've cloned the starter project, made customisations, and developed your app.

I'd like to suggest splitting the webpack configuration into modules that can be installed via NPM,  and merged using [webpack-merge](https://github.com/survivejs/webpack-merge) to produce the final configuration. I know that there have already been suggestions for splitting the configuration per environment, but here I'm talking about doing that per feature of the build.

For example, one might want to use Stylus and not Sass, and this kind of split would allow them to simply install the Stylus module instead of the Sass module.

I've already spent some time working on a project called [ngpack](https://github.com/ngpack/ngpack) to implement something like that. It mostly has the same configurations as `angular2-webpack` (copied actually), and the example app's [`webpack.config.js`](https://github.com/ngpack/ngpack/blob/master/examples/angular2-webpack-clone/webpack.config.js) doesn't have much in it, other than loading the ngpack, adding extensions and applying some customisations. I'd also like to point out the simplification of enabling HMR in [`app.module.ts`](https://github.com/ngpack/ngpack/blob/master/examples/angular2-webpack-clone/src/app/app.module.ts#L31), which is done with a simple `require` and a method call.

Basically, I'm trying to reduce the amount of build/dev-related code that is inside the main project repository, and move them out into installable NPM modules, in order to simplify the configuration file and allow for easier upgrades, while still making room for customisations to be made per project.

I personally think this is a better approach (obviously), so I'm going to ask if you would adopt this method? And if you wouldn't, I'm genuinely interested to know why, and I'm open to finding out better ways of improving the maintainability of build-related code in our projects.

**Comment by Foxandxss** [comment]
I see your point and it is a really really valid point, but the idea of a starter is just that, being a **starter**. You clone it for your app and you modify it to your needs. You are not really supposed to keep up to date with the starter anymore.

You can come to the changelog and say, ah, he added webpack 2, let's see how he did it so I can update mine.

I don't really expect that any application out there with my starter is going to use my configuration as is, because as you said, you probably want to make different decisions, so it is not really easy to come back and make updates.

I like your project, it is a good idea, but my goal with this starter is just different. A nice starting point for your app, not meant to be always up to date with your app.

Honestly, if what you need is a starter that is up to date with what you do, you should use the CLI.
